test/e2e/framework/nodes.go

```diff
@@ -0,0 +1,40 @@
+package framework
+
+import (
+	"strconv"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (f *Framework) FindGatewayAndNonGatewayNodes(cluster ClusterIndex) (gateway *v1.Node, nonGateway *v1.Node) {
+	nodes := AwaitUntil("list nodes", func() (interface{}, error) {
+		// Ignore the control plane node labeled as master as it doesn't allow scheduling of pods
+		return f.ClusterClients[cluster].CoreV1().Nodes().List(metav1.ListOptions{
+			LabelSelector: "!node-role.kubernetes.io/master",
+		})
+	}, NoopCheckResult).(*v1.NodeList)
+
+	var gatewayName, nonGatewayName string
+	for i := range nodes.Items {
+		if nodes.Items[i].Labels[GatewayLabel] == "true" {
+			gateway = &nodes.Items[i]
+			gatewayName = gateway.Name
+		} else {
+			nonGateway = &nodes.Items[i]
+			nonGatewayName = nonGateway.Name
+		}
+	}
+
+	Logf("Found gateway node: %s, non-gateway node: %s", gatewayName, nonGatewayName)
+	return
+}
+
+func (f *Framework) SetGatewayLabelOnNode(cluster ClusterIndex, nodeName string, isGateway bool) {
+	DoPatchOperation("/metadata/labels/submariner.io~1gateway", strconv.FormatBool(isGateway),
```

Comment:
is "~1" a typo? should be "/"?

Reply:
no typo - it's the special escape sequence for "/" (defined in some RFC) so it's not treated as part of the path.

Reply:
I added a comment and also used the `GatewayLabel` const with `strings.Replace` to make it clearer.